### ISSUE TITLE
New SplashScreen only gets displayed for a fraction of a second.

### ISSUE THREAD

**Issue by GAZ082** [FreeCAD/FreeCAD#16264]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Problem description

Looks like all the loading gets performed and after that the splash appears for a blink of an eye.

https://github.com/user-attachments/assets/ed066d97-e502-4a78-b62a-e463d5653db0

### Full version info

```shell
OS: Arch Linux (GNOME/gnome)
Word size of FreeCAD: 64-bit
Version: 0.22.0dev.38622 (Git) AppImage
Build type: Release
Branch: main
Hash: ed97f3c4591303ee3b54906768f8f4eba5211265
Python 3.11.9, Qt 5.15.13, Coin 4.0.2, Vtk 9.2.6, OCC 7.7.2
Locale: English/United States (en_US)
Installed mods: 
  * OpenTheme 2024.8.17
```

### Subproject(s) affected?

None

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by Syres916** [comment]
Confirmed, I re-enabled the messages just so I could see the SplashScreen and I'm using a 12 year old box so what it's like on a new piece of kit, must be close to invisible. A minimum length of display time of say one second wouldn't be unreasonable. @qewer33 did you not experience this scenario in testing?

**Comment by FEA-eng** [comment]
For me, it's working normally but I still have 0.21.2 and rather old computer.

```
OS: Windows 10 build 19045
Word size of FreeCAD: 64-bit
Version: 0.22.0dev.38622 (Git)
Build type: Release
Branch: main
Hash: ed97f3c4591303ee3b54906768f8f4eba5211265
Python 3.11.9, Qt 5.15.13, Coin 4.0.2, Vtk 9.2.6, OCC 7.7.2
Locale: Polish/Poland (pl_PL)
```

**Comment by qewer33** [comment]
Hmm yeah, it's because my PR disabled the splash messages which was unnecessarily taking time but ig it's way too fast on modern computers now... (On my machine it does take a second though but sometimes a little less)

So we have a couple of options:
- Bring back the splash messages
  - I gues we can do that...
- Set a fixed delay
  - Terrible idea *and also the internet would meme it to hell and back if it somehow gets discovered*.
- Ensure the splash screen shows for a minimum X amount of time, implemented via a timer or something
  - Still not very good but certainly a better idea than the last one since it'll be more complex and normal people won't be able to understand and meme it.
- Actually fix the scuffed startup routine, for some reason the splash screen shows a few seconds after FreeCAD starts and hides a few seconds earlier than when the main window appears. Also the only things that gets done while the splash is showing seems to be GUI initilization which as we can see now with the splash messages disabled, doesn't even take that long...
  -  Now the problem with this idea, we will need to mess with the startup routine and the base application code so it might cause a lot of platform specific bugs, not a very good idea to implement before the 1.0 release.

The third option seems like the best one but I'm open to all ideas. Please express your thoughts.

**Comment by GAZ082** [comment]
> This is really good, the quicker it gets to open, the better it is.

In theory. But in this case, the WBs and stuff gets loaded before the splash screen. So you are kept staring at the desktop, wondering what's going on and then boom.

**Comment by 3x380V** [comment]
No, WBs are loading while splash screen is shown, just see the code. With #16334 you can have messages back, so just store time splash was shown, compare with time it is about to disappear and eventually extend it. That's possible, since once Qt message loop is running, timers work.

**Comment by GAZ082** [comment]
@3x380V 

With "messages" you are referring to the notification that something is being loaded? Ie, the workbenches?

**Comment by 3x380V** [comment]
@GAZ082, see https://github.com/FreeCAD/FreeCAD/blob/fe40748c10503eed367840239db3e0f81873e359/src/Gui/StartupProcess.cpp#L430-L434
You might as well move showing splash earlier in the startup process, I just tried to put at the beginning of
https://github.com/FreeCAD/FreeCAD/blob/fe40748c10503eed367840239db3e0f81873e359/src/Gui/StartupProcess.cpp#L218-L220
But you need to check for side effects. Also OS needs some time to get image into memory and there is nothing application can do about.

**Comment by leoheck** [comment]
> In theory. But in this case, the WBs and stuff gets loaded before the splash screen. So you are kept staring at the desktop, wondering what's going on and then boom.

I see, however on my machine, the launch of Freecad is really quick now since 2 or 3 weekly builds ago.

**Comment by 3x380V** [comment]
We want do show splash only if `Gui::Application::runApplication()` not `Gui::Application::runApplication()` is about to be executed and only if it is not executed from Python. Is that correct?

**Comment by 3x380V** [comment]
Please build and test #16334. Whatever is written in that PR comment still apply.

**Comment by mnesarco** [comment]
Please consider my PR for 1.0, it is quite small and works until a bigger redesign of StartupProcess:
https://github.com/FreeCAD/FreeCAD/pull/16685

**Comment by GAZ082** [comment]
I can confirm that this is happening in the 1.1 branch, not the 1.0.RC

**Comment by luzpaz** [comment]
This is still happening, btw

**Comment by leoheck** [comment]
Honestly the best is to have the software as quickly as possible not the splash screen.

**Comment by GAZ082** [comment]
I like the splash, but if it's so hard to make it work as it should, so yes, let's scrap it.

**Comment by 3x380V** [comment]
@leoheck , agree, but FreeCAD does quite a lot while initializing. You could to lazy loading, etc. but that does not really solve much. It just shows main window faster, but user is still not able to interact as missing pieces are load on demand.

But back to the topic, see discussion in #16685. The problem is splash shown too late in the init process and too much is happening before `QGuiApplication` is instantiated. Having this fixed was prerequisite for my scrolling splashscreen PR, but as @wwmayer wanted to keep splash simple, I abandoned working on it.

Anyone wanting to continue is welcome to pick up #16334 and [soonsplash](https://github.com/3x380V/FreeCAD/tree/soonsplash) branch.

**Comment by luzpaz** [comment]
> But back to the topic, see discussion in #16685. The problem is splash shown too late in the init process and too much is happening before `QGuiApplication` is instantiated. Having this fixed was prerequisite for my scrolling splashscreen PR, but as @wwmayer wanted to keep splash simple, I abandoned working on it.

> Anyone wanting to continue is welcome to pick up #16334 and [soonsplash](https://github.com/3x380V/FreeCAD/tree/soonsplash) branch.

Any takers ?

Right now.. I see a splash screen that is totally black and then for a split second it gets populated by a legit graphic. Then it disappears. So in theory we have a black splash screen. 

https://github.com/user-attachments/assets/0419df25-504b-451d-a63c-9bccfa94a1a6

```
OS: Manjaro Linux (KDE/plasmawayland/wayland)
Architecture: x86_64
Version: 1.2.0dev.20251209 (Git shallow) AppImage
Build date: 2025/12/09 13:53:17
Build type: Release
Branch: grafted,grafted
Hash: 5ad6a1ba58c665c23f1d217c23e74379d8c1a5a4
Python 3.11.14, Qt 6.8.3, Coin 4.0.3, Vtk 9.3.1, boost 1_86, Eigen3 3.4.0, PySide 6.8.3
shiboken 6.8.3, xerces-c 3.3.0, IfcOpenShell 0.8.2, OCC 7.8.1
Locale: English/United States (en_US)
Navigation Style/Orbit Style/Rotation Mode: Touchpad/Turntable/Window center
Stylesheet/Theme/QtStyle: FreeCAD Dark.qss/FreeCAD Dark/
Logical DPI/Physical DPI/Pixel Ratio: 96/124.55/2
Installed mods: 
  * sheetmetal 0.7.58
  * Telemetry 1.0.4
```

**Comment by 3x380V** [comment]
@GAZ082, @leoheck, @luzpaz, I tagged #31176 as a fix for this issue, which it does with already merged #29917. Could you test whenever it fixes issue for you?

**Comment by GAZ082** [comment]
@3x380V latest dev working ok! thanks!

**Comment by mnesarco** [comment]
Main working great on linux mint (x11)
